### ISSUE TITLE
[REC] Holy Bulwark isn't showing up

### ISSUE THREAD

**Issue by ambonif** [Hekili/hekili#3550]
### Before You Begin

- [X] I confirm that I have downloaded the latest version of the addon.
- [X] I am not playing on a private server.
- [X] I checked for an [existing, open ticket](https://github.com/Hekili/hekili/labels/recommendation) for this issue and was not able to find one.
- [X] I edited the title of this issue (above) so that it describes the issue I am reporting.
- [X] I am reporting an issue with the default priority included with the specialization (imported or edited priorities are not supported).

### Describe the Issue

For Protection Paladins, Holy Bulwark/Sacred Weapon isn't shown or recommended at all. 

1. Charges doesn't seem to be detected correctly even if it's at 2 the addon conditions show 0.
2. The condition that doesn't require charges, still doesn't show the ability even if next_armament is holy_bulwark. I think next_armament just doesn't work correctly, because even if I use the ability and sacred_weapon is next, the correct value doesn't show.

![image](https://github.com/user-attachments/assets/50902b67-4b00-4483-9678-e6a4f2a294df)

### How to Reproduce

1. Enter the game as a Protection Paladin
2. Spec into Lightsmith
3. No conditions show Holy Bulwark

### Snapshot (Link)

https://pastebin.com/qF9wrRHu

**Comment by syrifgit** [comment]
You have your cooldowns shown on a separate display, so you need to provide a snapshot of that display, not primary.

**Comment by ambonif** [comment]
> You have your cooldowns shown on a separate display, so you need to provide a snapshot of that display, not primary.

I can move back Holy Bulwark back to Default/Primary, hold on.

**Comment by ambonif** [comment]
Primary - https://pastebin.com/s4R5jkRY
Cooldowns - https://pastebin.com/FdRjib3G

**Comment by syrifgit** [comment]
Can you show more than one button? It's selecting judgment and so it cuts off right before it shows the logic for the spell you're concerned about.

Make the display 3-4 buttons and grab the snapshot if you can

**Comment by ambonif** [comment]
Here's a lengthier one --

https://pastebin.com/YsEz8Yjd

**Comment by Hekili** [comment]
A test fix will be in 11.0.2-1.0.5

**Comment by ambonif** [comment]
> A test fix will be in 11.0.2-1.0.5

Heads up, it's still the same. If it matters, in the Abilities list, it is now "Sacred Weapon" and not "Holy Bulwark". Charges are being detected correctly, but it's still not being recommended in the actual rotation.

https://pastebin.com/PY699j7M

Relevant error looks to be saying that holy_armaments ability is unknown despite being talented into it.

**Comment by theninj4** [comment]
I'm having the same issue. After a little poking with snapshots and the priority, it knows I've got the talent:
```
holy_armaments = 1/1
```
but won't recommend the button because "ability unknown":
```
        Current recommendation was NO ACTION at +10.00s.
        
        1.   holy_armaments ( precombat - 1 ) - ability unknown
        Time spent on this action:  0.10ms
        TimeData:Protection Paladin-precombat-1:holy_armaments:0.10:Ability Known, Enabled(0.10)
```

**Comment by theninj4** [comment]
Edit: Sorry for the rambling debugging. This diff gets the button working:
```
+    if talent.holy_armaments.enabled then
+        if IsActiveSpell( 432472 ) then applyBuff( "sacred_weapon_ready" )
-    if talent.holy_bulwark.enabled then
-        if IsActiveSpell( 432478 ) then applyBuff( "sacred_weapon_ready" )
         else applyBuff( "holy_bulwark_ready" ) end
     end
```

I've had to edit my rotation, this error is throwing me. The expression evaluates to true, then the criteria fails?
> actions+=/holy_armaments,if=next_armament=sacred_weapon
```
        12.  holy_armaments ( default - 8 )
        The action (holy_armaments) is usable at (0.00 + 0.00).
         - the action is ready before the current recommendation (at +0.00 vs. +10.00).
         - this entry's criteria FAILS: next_armament[sacred_weapon] = sacred_weapon[true]
```

If you still don't see the ability, it's because the sacred-weapon only shows up with at least 4 targets. I edited that in the UI down to 0.

**Comment by EricMiukyQin** [comment]
> A test fix will be in 11.0.2-1.0.5

Tested after v11.0.2 1.0.5b, this issue still exists. I recognized the problem may be in actionlist conditions. next_armament is a string while sacred_weapon is a boolean, which would never be equal with each other, so that holy bulwark spell would never be recommended.
![image](https://github.com/user-attachments/assets/292ab76f-55b5-4885-b621-5ec91c4d38c0)
